corehq/apps/reports/view_helpers.py

```diff
@@ -25,15 +25,11 @@ def case_hierarchy_context(case, timezone=None):
                 parent_case = idx.referenced_case
             except ResourceNotFound:
                 parent_cases.append(None)
-                last_parent_id = None
             else:
                 parent_case.index_info = _index_to_context(idx, is_ancestor=True)
                 parent_cases.append(parent_case)
-                last_parent_id = parent_case.case_id
 
-        for c in descendant_case_list:
-            if not getattr(c, 'treetable_parent_node_id', None) and last_parent_id:
-                c.treetable_parent_node_id = last_parent_id
+        case.treetable_parent_node_id = parent_cases[-1].case_id if parent_cases[-1] else None
```

Comment:
Won't `parent_cases[-1]` fail if `parent_cases` is empty?

Reply:
`parent_cases` can't be empty - this is all inside a `if case.live_indices`, and something is added to `parent_cases` for every live index.  It does feel like there's still room to make this whole thing clearer though, and I'm open to suggestions if you've got an approach you like.

Reply:
ah, this makes sense. I misread the indentation, I raised this because I thought it was outside of the if.